swap_meet/vendor.py

```diff
@@ -1,2 +1,83 @@
+from .item import Item
+
+
 class Vendor:
-    pass
+    def __init__(self, inventory=None):
+        if inventory:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        items_by_category = []
+        for item in self.inventory:
+            if item.category == category:
+                items_by_category.append(item)
+
+        return items_by_category
+
+    def swap_items(self, friend, my_item, their_item):
+
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.remove(my_item)
+            self.add(their_item)
+            friend.remove(their_item)
+            friend.add(my_item)
+            return True
+        else:
+            return False
```

Comment:
Notice how top heavy this function looks? We can make it look more balanced by using a guard clause.

```suggestion
    def swap_items(self, friend, my_item, their_item):

        if my_item not in self.inventory and their_item not in friend.inventory:
            return False
            
        self.remove(my_item)
        self.add(their_item)
        friend.remove(their_item)
        friend.add(my_item)
        return True
```